### PR TITLE
improve nix highlights and injections

### PR DESCRIPTION
**Pull request by figsoda**
follow up to #3842, now that I'm more familiar with this

### CODE REVIEW

File: queries/nix/highlights.scm [hunk 1]
```diff
@@ -62,33 +65,122 @@
...
-; import
-((identifier) @_i (#eq? @_i "import")) @include
-
-; null
-((identifier) @_i (#eq? @_i "import")) @constant.builtin
-
-; these are technically functions but they act more like keywords (abort and throw are control flow, derivation is a core language construct)
-((identifier) @_i (#match? @_i "^(abort|derivation|throw)$")) @keyword
+; modified from `nix __dump-builtins | jq keys`
+((identifier) @function.builtin (#any-of? @function.builtin
```
Comment (figsoda) [inline queries/nix/highlights.scm:78]: if this is too much I can just keep the ones that are not directly available without `builtins`, which is like 1/4 of those

Comment (rummik) [reply queries/nix/highlights.scm:78]: Technically the remaining ones are available with a `__` prefix iirc.  ex: `builtins.readDir` is accessible via `__readDir`

Comment (figsoda) [reply queries/nix/highlights.scm:80]: I added all the builtins/primops I was able to find

File: queries/nix/highlights.scm [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
+  "in"
+  "inherit"
+  "let"
+  "rec"
+  "with"
 ] @keyword
 
+; these are technically functions but they act more like keywords (abort and throw are control flow, derivation is a core language construct)
+((identifier) @keyword
+  (#any-of? @keyword "abort" "derivation" "throw"))
```
Comment (theHamsta) [inline queries/nix/highlights.scm:13]: abort/throw `@expection`?

**Review by theHamsta — COMMENTED** [review]
This looks overall good! Please see my small comment.

File: queries/nix/highlights.scm (hunk 2 above)
Comment (figsoda) [reply queries/nix/highlights.scm]: fixed it

**Comment by figsoda** [comment]
@theHamsta do you mind giving this another review? Thanks!

**Comment by theHamsta** [comment]
Thank you!

**Review by theHamsta — APPROVED** [review]

**Pull request merged**